integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,269 @@
 	configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
-	agentRuntimeMinutes = 5 //20min desired but 5mins for testing purposes 
+	agentRuntimeMinutes = 5 //20 mins desired but 5 mins for testing purposes
 	DynamoDBDataBase = "CWAPerformanceMetrics"
-
+	testLogNum = "PERFORMANCE_NUMBER_OF_LOGS"
 )
 
+type LogInfo struct {
+	FilePath string `json:"file_path"`
+	LogGroupName string `json:"log_group_name"`
+	LogStreamName string `json:"log_stream_name"`
+	Timezone string `json:"timezone"`
+}
+
 func TestPerformance(t *testing.T) {
+	//get number of logs for test from github action
+	//logNum, err := strconv.Atoi(os.Getenv(testLogNum)) //requires a commit from Okan that updates the workflow file so the log tests will run concurrently
+	// if err != nil {
+	// 	t.Fatalf("Error: cannot convert test log number to integer, %v", err)
+	// }
+	logNum := 10 //THIS IS TEMPORARY SO CODE RUNS
+
+	
+
+	configFilePath, err := GenerateConfig(logNum)
+	if err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+
+	log.Printf("config generated at %s\n", configFilePath)
+	defer os.Remove(configFilePath)
+
+	tpsVals := []int {
+		10,
+		100,
+		1000,
+	}
+	
 	agentContext := context.TODO()
 	instanceId := test.GetInstanceId()
 	log.Printf("Instance ID used for performance metrics : %s\n", instanceId)
 
-	test.CopyFile(configPath, configOutputPath)
+	//data base
+	dynamoDB := InitializeTransmitterAPI(DynamoDBDataBase) //add cwa version here
+	if dynamoDB == nil {
+		t.Fatalf("Error: generating dynamo table")
+	}
+
+	//run tests
+	for _, tps := range tpsVals {
+		t.Run(fmt.Sprintf("TPS run: %d", tps), func(t *testing.T) {
+			test.CopyFile(configFilePath, configOutputPath)
 
-	test.StartAgent(configOutputPath, true)
+			test.StartAgent(configOutputPath, true)
 
-	agentRunDuration := agentRuntimeMinutes * time.Minute
-	//let agent run before collecting performance metrics on it
-	time.Sleep(agentRunDuration)
-	log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
-	test.StopAgent()
+			agentRunDuration := agentRuntimeMinutes * time.Minute
 
-	//collect data
-	data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext)
-	//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+			err := StartLogWrite(agentRunDuration, configFilePath, tps)
+			if err != nil {
+				t.Fatalf("Error: %v", err)
+			}
+
+			log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
+			test.StopAgent()
+
+			//collect data
+			data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext, configFilePath)
+			//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+			if err != nil {
+				t.Fatalf("Error: %v", err)
+			}
+
+			if data == nil {
+				t.Fatalf("No data")
+			}
+			
+			//append test metadata so we can differentiate between tests
+			data, err = AppendTestMetadata(data, logNum, tps)
```

Comment:
I thought about doing that, but the data is returned in a struct which needs to be marshalled into a json before the metadata gets added so that I don't need to define and create an entire new custom struct. I left it in the test file because that's where the metadata is generated so I felt it was more relevant